idaes/core/initialization/initializer_base.py

```diff
@@ -0,0 +1,375 @@
+        """
+        Load initial guesses for variables into model.
+
+        Args:
+            model: Pyomo model to be initialized.
+            initial_guesses: dict of initial guesses to load.
+            json_file: file name of json file to load initial guesses from as str.
+
+        Note - can only provide one of initial_guesses or json_file.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError if both initial_guesses and json_file are provided.
+        """
+        if initial_guesses is not None and json_file is not None:
+            self.status = InitializationStatus.Error
+            raise ValueError(
+                "Cannot provide both a set of initial guesses and a json file to load."
+            )
+
+        if initial_guesses is not None:
+            self._load_values_from_dict(model, initial_guesses)
+        elif json_file is not None:
+            # TODO: What should we load here?
+            # As this is just initialization, for now I am only loading variable values
+            from_json(
+                model, fname=json_file, wts=StoreSpec().value(only_not_fixed=True)
+            )
+        else:
+            self.get_logger(model).info_high(
+                "No initial guesses provided during initialization."
+            )
+
+    def fix_initialization_states(self, model: Block):
+        """
+        Call to model.fix_initialization_states method. Method will pass if
+        fix_initialization_states not found.
+
+        Args:
+            model: Pyomo Block to fix states on.
+
+        Returns:
+            None
+        """
+        try:
+            model.fix_initialization_states()
+        except InitializationError:
+            pass
+
+    def precheck(self, model: Block):
+        """
+        Check for satisfied degrees of freedom before running initialization.
+
+        Args:
+            model: Pyomo Block to fix states on.
+
+        Returns:
+            None
+
+        Raises:
+            InitializationError if Degrees of Freedom do not equal 0.
+        """
+        if not degrees_of_freedom(model) == 0:
+            self.status = InitializationStatus.DoF
+            raise InitializationError(
+                f"Degrees of freedom for {model.name} were not equal to zero during "
+                f"initialization (DoF = {degrees_of_freedom(model)})."
+            )
+
+    def initialization_routine(self, model: Block):
+        """
+        Placeholder method to run initialization routine. Derived classes should overload
+        this with the desired routine.
+
+        Args:
+            model: Pyomo Block to initialize.
+
+        Returns:
+            Overloaded method should return a Pyomo solver results object is available,
+            otherwise None
+
+        Raises:
+            NotImplementedError
+        """
+        self.status = InitializationStatus.Error
+        raise NotImplementedError()
+
+    def restore_model_state(self, model: Block):
```

Comment:
I originally had that but removed it (one I stored the state on the initializer). If we let the user provide a state there is no guarantee that it would be the state we are expecting, but then again it would be an option argument so user beware?

Reply:
This is just a method that may be useful in other contexts, e.g. restoring the model to some intermediate state in a complicated initialization routine.

Reply:
I would be inclined to say it would be better to use the serialization tools directly in that case - this was deliberately written as a limited use case of restoring the initial state.